qutebrowser/config/config.py

```diff
@@ -642,7 +644,11 @@ def init(parent=None):
     """
     configdata.init()
 
-    yaml_config = configfiles.YamlConfig()
+    no_autoconfig = os.path.exists(
+        os.path.join(standarddir.config(), 'no-autoconfig'))
+    # Not creating a YamlConfig instance prevents changes from being written
+    # back to the autoconfig
+    yaml_config = None if no_autoconfig else configfiles.YamlConfig()
```

Comment:
Setting this to `None` currently has some bad consequences, when we set via the UI, we get a crash:

```
Traceback (most recent call last):
  File "/home/jay/Code/qutebrowser/qutebrowser/commands/runners.py", line 313, in run_safely
    self.run(text, count)
  File "/home/jay/Code/qutebrowser/qutebrowser/commands/runners.py", line 292, in run
    result.cmd.run(self._win_id, args, count=count)
  File "/home/jay/Code/qutebrowser/qutebrowser/commands/command.py", line 516, in run
    self.handler(*posargs, **kwargs)
  File "/home/jay/Code/qutebrowser/qutebrowser/config/config.py", line 275, in set
    self._set_next(option, values, temp=temp)
  File "/home/jay/Code/qutebrowser/qutebrowser/config/config.py", line 291, in _set_next
    self._config.set_str(option, values[0], save_yaml=not temp)
  File "/home/jay/Code/qutebrowser/qutebrowser/config/config.py", line 464, in set_str
    self._yaml.values[name] = converted
AttributeError: 'NoneType' object has no attribute 'values'
```

But we do want to handle this case differently (`:set` while this option is available)